iceoryx_posh/source/roudi/roudi.cpp

```diff
@@ -254,6 +263,45 @@ void RouDi::processRuntimeMessages(runtime::IpcInterfaceCreator&& roudiIpcInterf
     IOX_LOG(INFO, "RouDi is ready for clients");
     fflush(stdout); // explicitly flush 'stdout' for 'launch_testing'
 
+    /*
+     * We get information about how they are running. If as a unit, then we launch
+     * watchdog and send a notification about the launch, otherwise we do nothing
+     */
+    const char* invocation_id = std::getenv("INVOCATION_ID");
+    if (invocation_id != nullptr)
+    {
+        IOX_LOG(WARN, "Run APP in unit(systemd)");
+        listen_thread_watchdog = std::thread([this] {
+            if (auto wdres = sd_notify(0, "READY=1") < 0)
+            {
+                std::array<char, SIZE_ERROR_MESSAGE> buf{};
+                strerror_r(-static_cast<int>(wdres), buf.data(), buf.size());
+                IOX_LOG(ERROR, "WatchDogError: " << std::string(buf.data()));
+                return;
+            }
+            IOX_LOG(DEBUG, "WatchDog READY=1");
+
+            IOX_LOG(INFO, "Start watchdog");
+            while (m_runHandleRuntimeMessageThread.load())
+            {
+                if (auto wdres = sd_notify(0, "WATCHDOG=1") < 0)
```

Comment:
Please use the `IOX_POSIX_CALL`. It takes care of the error handling

Reply:
> Please use the `IOX_POSIX_CALL`. It takes care of the error handling

@elBoberido fixed